### PR TITLE
docs(core): Updated doc to encourage use of non-deprecated API instead of the deprecated ComponentFactory

### PR DESCRIPTION
**Pull request by stevekacou**
ComponentFactory class is deprecated, updated doc to encourage use of non-deprecated API.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by mlc-mlapis** [comment]
Would it be nice to mention the cases, where `ComponentFactoryResolver` is still required to use?
```
const domPortalOutlet = new DomPortalOutlet(
			anchor,
			this._componentFactoryResolver,
			this._appRef,
			customInjector
		);
const componentPortal = new ComponentPortal<LazyEditorComponent>(ModuleLazyEditorComponent);
const componentPortalRef = domPortalOutlet.attachComponentPortal(componentPortal);
```

File: packages/core/src/linker/component_factory.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ import {NgModuleRef} from './ng_module_factory';
 import {ViewRef} from './view_ref';
 
 /**
- * Represents a component created by a `ComponentFactory`.
+ * Represents a component created by the `createComponent` method of ViewContainerRef class or by the depreciated `ComponentFactory` class .
  * Provides access to the component instance and related objects,
  * and provides the means of destroying the instance.
```
Comment (AndrewKushnir) [inline packages/core/src/linker/component_factory.ts:21]: I think we can just drop the `ComponentFactory` reference, here is a proposal:
```suggestion
 * Represents a reference to an instantiated component.
 * Provides an access to the component instance itself, as well as related objects,
 * and provides the method to destroy the component instance.
```

Comment (pkozlowski-opensource) [reply packages/core/src/linker/component_factory.ts:21]: Agree with @AndrewKushnir suggestion.

**Comment by pkozlowski-opensource** [comment]
@stevekacou if you would like to take this PR forward you would need to sign the CLA and address [Andrew's suggestion](https://github.com/angular/angular/pull/46312#pullrequestreview-1000471654)

We can't merge this PR without the mentioned changes.

**Comment by pkozlowski-opensource** [comment]
@stevekacou as we are trying to keep our PR queue small, I'm going to close this PR for now as it needs additional work. Feel free to re-open it when you've got time to restart work on it.

**Pull request closed without merging**